### PR TITLE
Deployment updates

### PR DESCRIPTION
**Pull request by alexdrenea**
### Deployment script
- Simplifying Parameters for deployment script:
    - All SQL related parameters are optional and populated with defaults for simplicity and security.
    - Resource group name is optional and can be inferred from the web app prefix
    - ARM template default is the local ./deploy.json
- replace MSDeploy in ARM with PowerShell script Deploy
    - this fixes issue where cannot run the deployment script locally (#95)
    - simplifies deployment but removing the need to push zip packages to storage blob
    - clears the path for using Linux Web App in the future (the MSDeploy extension is only available for windows)
- Other updates
    - Replace `get-AzureRMContext` command with the newer AZ equivalent - `Get-AzContext` -> confirmed this works the same in the cloud shell
    - Updated method of creating the landingPageAppRegistration by using the `New-AzureADApplication` -> this removes the dependency on the `jq` utility that is not installed by default on users' machines

### ARM template
- Add KeyVault support
    - Arm template deploys a KeyVault
    - Web app get managed identity with access to KeyVault
    - Connection string to SQL and AAD secret get stored in KeyVault and referenced in Web App via App Service KeyVault References for optimal security
- Other
    - Fix SLQ firewall setting (#170)
    - Cleanup some variables and parameters in the ARM template.
    - Removed MSDeploy extension since it causes issues with publishing when not using correct `CompressArchive` version

### CODE REVIEW

**Review by dstarr — COMMENTED** [review]
Not sure I am comfortable inferring the resource group name from the resource prefix. That seems like it would produce potentially unexpected behavior. @code4clouds what do you think?

**Comment by alexdrenea** [comment]
That's fair. We can turn that back into a mandatory parameter. 
But it's still an option to provide it if users want to....Maybe we can look at adjusting the naming convention? I noticed a lot of folks (as well as the official MS docs) recommend pre-pending the Resource Group Name with "rg"

Open to make the change(s) either way

**Comment by dstarr** [comment]
Nah, it's cool. Not that big a deal given it's optional. I'll try to run
the installer tomorrow.

File: deployment/Templates/deploy.json [hunk 1]
```diff
@@ -79,29 +79,52 @@
...
     "appServicePlanName": "[concat(parameters('webAppNamePrefix'),'AmpSvcPlan')]",
     "ProvisioningServiceAppName": "[concat(parameters('webAppNamePrefix'),'-portal')]",
     "PublisherServiceAppName": "[concat(parameters('webAppNamePrefix'),'-admin')]",
-    "collation": "SQL_Latin1_General_CP1_CI_AS",
+    "KeyVaultName": "[toLower(concat(parameters('webAppNamePrefix'),'-kv'))]",
     "sqlDBName": "AMPSaaSDB"
   },
   "resources": [
+    {
+      "type": "Microsoft.KeyVault/vaults",
```
Comment (santhoshb-msft) [inline deployment/Templates/deploy.json:91]: Should KeyVault be optional?

Comment (code4clouds) [reply deployment/Templates/deploy.json:91]: Yes, KeyVault should be optional.

**Comment by alexdrenea** [comment]
Happy to see this reviveved. Happy to finish the work on this if need be. Just let me know what needs to be done. 
@code4clouds , @msalemcode let me know if you want to book a chat to talk though this.

**Comment by code4clouds** [comment]
We will follow this PR on #279

**Pull request closed without merging**